src/sage/groups/libgap_mixin.py

```diff
@@ -963,3 +965,128 @@ def is_isomorphic(self, H):
             (False, False, False)
         """
         return self.gap().IsomorphismGroups(H.gap()) != libgap.fail
+
+
+def minimum_generating_set(group) -> list:
+    """
+    Return a list of the minimum generating set of ``group``. Elements of the list are
+    objects of GAP.
+
+    INPUT:
+
+    - ``group`` -- a group
+
+    OUTPUT:
+
+    A list of elements that generate the group. Elements of the list are objects of GAP.
+
+    .. SEEALSO::
+
+        :meth:`sage.groups.group.Group.minimum_generating_set`
+
+    ALGORITHM:
+
+    We follow :doi:`10.1016/j.jalgebra.2023.11.012` (:arxiv:`2306.07633`).
+
+    TESTS:
+
+        Test that the resultant list is able to generate the original group::
+
+            sage: from sage.groups.libgap_mixin import minimum_generating_set
+            sage: p = libgap.eval("DirectProduct(AlternatingGroup(5),AlternatingGroup(5))")
+            sage: s = minimum_generating_set(p); s
+            [(3,4,5)(8,9,10), (1,2,3,4,5)(6,7,8)]
+            sage: set(p.AsList()) == set(libgap.GroupByGenerators(s).AsList())
+            True
+
+        Test that elements of resultant list are GAP objects::
+
+            sage: from sage.groups.libgap_mixin import minimum_generating_set
+            sage: G = PermutationGroup([(1,2,3), (2,3), (4,5)])
+            sage: s = minimum_generating_set(G); s
+            [(2,3), (1,3,2)(4,5)]
+            sage: s[0].parent()
+            C library interface to GAP
+    """
+    if not isinstance(group, GapElement):
+        try:
+            group = group._libgap_()
+        except AttributeError:
+            raise NotImplementedError("only implemented for groups that contain a working __libgap__ mehtod")
+
+    if not group.IsFinite().sage():
+        raise NotImplementedError("only implemented for finite groups")
+
+    # A function to check if the group is generated by the given generators or not.
+    def is_groupbygens(group, gens):
```

Comment:
```suggestion
    def is_group_by_gens(group, gens):
```
Better to follow standard Python conventions.